docs/table-upgrade.md

```diff
@@ -0,0 +1,41 @@
+# Table Upgrade logic and data structures
+The Hive Metastore migration process will upgrade the following Assets:
+- Tables ond DBFS root
+- External Tables
+- Views
+
+We don't expect this process to be a "one and done" process. This typically is an iterative process and may require a few runs.
+
+We suggest to keep track of the migration and provide the user a continuous feedback of the progress and status of the upgrade.
+
+Here is the outline of the process:
+1. The assessment runs and capture all the tables in HMS (Done).
+   1. Each table is categorized based on the type and storage
+   1. We use the "upgraded_to" table property to determine if the table was already upgraded to UC
+1. The assessment generates a list of "recommended External Locations". These locations are used by the external tables and are required for an "in place" upgrade.
+1. The assessment generates "databases" tables or an outline for a configuration file.
+1. It can be used by the user to configure the database upgrade.
```

Comment:
Is the intention here to create a configuration skeleton that the user could ultimately choose to not further configure? I.e. I want this table to go here, that table to go there, this database to be in this catalog, and renamed, etc.

If that's the case, that is a minimal action, but if they have to further configure after running this that adds additional complexity.

This seems unwieldy to do in a configuration file. Knowing how analysts like to work (as this would 100% become an analyst job), it would probably need to be exportable into a CSV so it can be exported into one of our least favorite spreadsheet products to be modified before being imported back in.

Reply:
Think of this as

0. I want to upgrade all my tables
1. I run a job, it generates a manifest of all my tables
2. I can optionally use that manifest and things are sort of automatically upgraded, with best guess options, i.e. move the table to a workspace catalog
3. I can choose to modify the manifest (by making the manifest human readable) and configure where my tables will go, and then I put the manifest back somewhere
4. I can then run another job, and it creates all my tables, as well as *another* job to automatically sync these tables for upstream changes from HMS that I can configure and run.

Reply:
That's exactly what I'm recommending.
I think the only open question is what medium do we use for the Manifest.

Reply:
Documented